app/forms/form_base.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+class FormBase
+  include ActiveModel::Model
+  include ActiveModel::AttributeAssignment
+  include ActiveModel::Validations::Callbacks
+
+  attr_accessor :trainee, :user, :params, :store, :new_attributes, :fields
+
+  delegate :id, :persisted?, to: :trainee
+
+  after_validation :track_validation_error, if: -> { errors.any? }
+
+  def initialize(trainee, params: {}, user: nil, store: FormStore)
```

Comment:
Thinking `trainee` argument should also be a keyword argument.

Reply:
Turns out making trainee a keyword argument opens a can of worms. @gpeng and I had a quick chat about it and decided to leave for another day since it breaks too many specs and we want to get this piece of the work done today.

I suspect that we won't be making trainee a keyword argument in the future because we have forms that instantiate other forms (see `Diversities::DisabilityDetailForm` for example), so it makes sense to have `trainee` a none keyword argument because all others are optional. It also makes it easy to pass the trainee instance into other objects by direct reference. Anyway, a discussing for another day.